### PR TITLE
Beepsky can be fully rebuilt and drops a special baton

### PR DESCRIPTION
**Pull request by Superlagg**
[FEATURE]
## About the PR 

If Beepsky dies, two things now happen: his wreckage is flagged as his and he drops a new type of stun baton.

When his wreck is reassembled with this baton, he's fully rebuilt, complete with attack_on_move properly set.

This securitron stun baton can also be used in crafting ordinary non-Beepsky secbots. This gives us a secbot that can attack_on_move, albeit at 1/3 of Beepsky's rate (25%/tile vs 75%/tile).

The baton can also be scanned by mechanics (like all batons), so you can make more sorta-robust secbots. Just gotta kill Beepsky first.

Also, once the secbot assembly has an arm, you can complete the bot by giving it a stun baton. It'll even drop that baton when it dies! Not sure why you'd want to, non-securitron batons don't give the bot anything special over the type Beepsky drops.

It also fixes the bug I left in from the last secbot PR where the scrap secbots are supposed to drop werent given a location. Whoops.

## Why's this needed? 

Right now, if Beepsky is killed and rebuilt, he won't be as robust as he used to be. Plus you can't even use his own baton to build him!

## Changelog

```
(u)Superlagg:
(*)Beepsky now drops his own personal stun baton, and he can now be properly rebuilt when he dies.
(+)Beepsky's baton can be used to build other non-Beepsky securitrons, granting them some of his robustness.
```

### CODE REVIEW

**Comment by pali6** [comment]
Shouldn't /obj/machinery/bot/secbot/beepsky have is_beepky set to 3?

**Comment by Superlagg** [comment]
> Shouldn't /obj/machinery/bot/secbot/beepsky have is_beepky set to 3?

That's what I get for not using defines. I'll take a look!

**Comment by Superlagg** [comment]
Added some defines, cleaned up some of the drop-loot section, and boosted the dropped baton's contraband level if it was Beepsky's (and had the assembly check this var to see if it should give it the unique name).

Thanks for getting me to look at this PR again, never would've caught the fact that normal secbots armed with generic special secbot batons would drop named batons, despite the baton never having entered Beepsky's possession

File: code/modules/robotics/bot/secbot.dm [hunk 1]
```diff
@@ -1,3 +1,9 @@
+#define IS_NOT_BEEPSKY_AND_HAS_SOME_GENERIC_BATON 0				// Just some everyday bot on the beat
+#define IS_BEEPSKY_AND_HAS_HIS_SPECIAL_BATON 1						// Full-assed Beepsky
+#define IS_NOT_BEEPSKY_BUT_HAS_HIS_SPECIAL_BATON 2				// A Beepsky brand secboton
+#define IS_BEEPSKY_BUT_HAS_SOME_GENERIC_BATON 3						// A generic-ass shitcurity baton
+#define IS_NOT_BEEPSKY_BUT_HAS_A_GENERIC_SPECIAL_BATON 4	// A generic, non-Beepsky brand secboton
```
Comment (pali6) [inline code/modules/robotics/bot/secbot.dm:5]: I feel like this sort of thing would be better either as two variables (is_beepsky, baton_type) or through bit flags in one variable but it's ok even like this to some extent.

File: code/modules/robotics/bot/secbot.dm [hunk 2]
```diff
@@ -1185,6 +1198,52 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
...
+				qdel(W)
+			else												// On any other securitron assembly?
+				boutput(user, "You give the [src] [W] and connect a cable in the arm to the baton's parallel port, completing the Securitron! Beep boop.")
+				var/obj/machinery/bot/secbot/S = new /obj/machinery/bot/secbot(get_turf(src))
+				S.beacon_freq = src.beacon_freq
+				S.hat = src.hat
+				S.name = src.created_name		// We get an upgraded securitron
+				S.attack_per_step = 2				// 25% chance to attack_on_move, as opposed to 75%
+				S.loot_baton_type = W.type	// So we can drop it all over again.
+				if (W.contraband != 4)
```
Comment (pali6) [inline code/modules/robotics/bot/secbot.dm:1217]: Using a contraband to check if it's the real deal (I think that's what this is?) is not a great solution especially considering that:
a.) someone might change the baton's contraband level and not know about this bit of code thus breaking it
b.) it's not clear at all what is going on here
I'd probably add a var on the beepsky baton that says if it's the original or a copy.

Comment (Superlagg) [reply code/modules/robotics/bot/secbot.dm]: Done!

File: code/modules/robotics/bot/secbot.dm (hunk 1 above)
Comment (Superlagg) [reply code/modules/robotics/bot/secbot.dm:5]: I don't really know how bitflags work, when I do, I'll definitely think about using them!

**Pull request merged**